lib/BaseClient.ts

```diff
@@ -45,28 +45,54 @@ abstract class BaseClient extends EventEmitter {
    * Returns the total balance available across all channels.
    */
   public abstract channelBalance(): Promise<ChannelBalance>;
-  protected setStatus(status: ClientStatus): void {
+
+  protected setStatus = async (status: ClientStatus): Promise<void> => {
     this.logger.info(`${this.constructor.name} status: ${ClientStatus[status]}`);
     this.status = status;
-    this.checkTimers();
+    await this.setTimers();
   }
-  private checkTimers() {
+
+  private setTimers = async () => {
     if (this.status === ClientStatus.ConnectionVerified) {
-      this.updateCapacityTimer = setInterval(async () => {
-        try {
-          this.maximumOutboundCapacity = (await this.channelBalance()).balance;
-        } catch (e) {
-          // TODO: Mark client as disconnected
-          this.logger.error(`failed to fetch channelbalance from client: ${e}`);
-        }
-      }, this.CAPACITY_REFRESH_INTERVAL);
+      if (!this.updateCapacityTimer) {
+        await this.updateCapacity();
+        this.updateCapacityTimer = setInterval(this.updateCapacity, BaseClient.CAPACITY_REFRESH_INTERVAL);
+      }
+
+      if (this.reconnectionTimer) {
+        clearTimeout(this.reconnectionTimer);
+        this.reconnectionTimer = undefined;
+      }
     } else {
       if (this.updateCapacityTimer) {
         clearInterval(this.updateCapacityTimer);
+        this.updateCapacityTimer = undefined;
       }
+      if (this.status !== ClientStatus.Disabled) {
+        if (!this.reconnectionTimer) {
+          this.reconnectionTimer = setTimeout(this.verifyConnection, BaseClient.RECONNECT_TIMER);
+        } else {
+          this.reconnectionTimer.refresh();
```

Comment:
If we decide to use the [refresh API](https://nodejs.org/api/timers.html#timers_timeout_refresh) we'll also have to update the:
[mininum required nodejs version](https://github.com/ExchangeUnion/xud/blob/master/package.json#L182) to >=v10.2.0 and make sure we also update our simnet etc. machines. Using an older version of nodejs will crash my xud process. Otherwise the fix is working fine.

Reply:
Alternative solution (uglier, but less work) would be to just clear the timeout before setting a new one.

Reply:
You're right although I'm OK with going to node 10 as the minimum version, given that 8 is only going to be maintained through the end of the year. I'll open another PR to update the minimum version in package.json.

Reply:
I actually just went ahead and put it in this PR.